packages/salesforcedx-vscode-core/src/orgPicker/orgList.ts

```diff
@@ -54,25 +54,7 @@ export class OrgList implements vscode.Disposable {
     const authFilesArray = await AuthInfo.listAllAuthorizations().catch(
       err => null
     );
-
-    if (authFilesArray === null || authFilesArray.length === 0) {
-      return null;
-    }
-    const authInfoObjects: FileInfo[] = [];
-    for (const authFile of authFilesArray) {
-      try {
-        const filePath = path.join(
-          await ConfigFile.resolveRootFolder(true),
-          '.sf',
-          authFile.username
-        );
-        const fileData = readFileSync(filePath, 'utf8');
-        authInfoObjects.push(JSON.parse(fileData));
-      } catch (e) {
-        console.log(e);
-      }
-    }
-    return authInfoObjects;
+    return authFilesArray;
```

Comment:
:|

Reply:
Our own @AnanyaJha made this change (well a version of it) [here](https://github.com/forcedotcom/salesforcedx-vscode/pull/890/files]

@AnanyaJha any memory about why we'd need to get the list of authorizations from global from core, then read the json files for each of those vs just using what they provide for us?

Reply:
Ok so I started feeling real paranoid about this so I looked at the types returned from each of these: 
```
export declare type OrgAuthorization = { // new structure
    orgId: string;
    username: string;
    oauthMethod: 'jwt' | 'web' | 'token' | 'unknown';
    aliases: Nullable<string[]>;
    configs: Nullable<string[]>;
    isScratchOrg?: boolean;
    isDevHub?: boolean;
    isSandbox?: boolean;
    instanceUrl?: string;
    accessToken?: string;
    error?: string;
    isExpired: boolean | 'unknown';
};
export interface FileInfo { // Old structure
    scratchAdminUsername?: string;  // Doesn't exist but isScratchOrg does
    isDevHub?: boolean;
    username: string;
    devHubUsername?: string; // doesn't exist 
    expirationDate?: string;  // Doesn't exist but isExpired does
  }

Reply:
this kind of blew my mind...I was like...these types don't match...how is the even compiling?

Reply:
I guess it's b/c the required properties for the two interfaces overlap

Reply:
previously line 70 did a JSON.parse which returns an any so that made sense (not logically but for compilation)

Reply:
FYI, I Took another look at this and I was able to make a much smaller change that got it working without changing the shape of the returned data.  Had to point this function to the correct directory where these auth files are stored, which might have shifted slightly with the upgrade to core v3.  That PR is much cleaner.  See the change here: https://github.com/forcedotcom/salesforcedx-vscode/pull/4290/files#diff-2b72c15ab3c2e3d66f269772e80aed9902d2d85d9acb2a462116d03541484d4fR66-R67